### ISSUE TITLE
Feature request: Bookmark/Star function

### ISSUE THREAD

**Issue by klasrocket** [vranki/siilihai-client#7]
Guess there is no better place to put a feature request.

I've tried myself to implement a bookmark function but got stuck. Tried to more or less clone the read/unread functionality. Would be best to have the opportunity to bookmark either group, thread or message but would make sense to only have thread or message as well. Some kind of vier filter would be nice as well to show only bookmarked items but not necessary.

**Comment by vranki** [comment]
Yes, this is a good place for feature requests. 

I've been planning for long time to "watch" a thread, which would mean that the thread would be highlighted if it would have new messages. Would this be about the same as bookmarking function?

**Comment by klasrocket** [comment]
Yes, that sounds to be about the same idea as mine. I guess the best would be free text tags, assignable to all items; groups, threads and messages and some kind of filter/search.
But a boolean called watch or bookmarked would be really useful! Will try to put some more effort into it but still new to c++ and qt. 

Was trying to use the tag/label function here to mark as feature request but guess it only works for repo owners.

**Comment by vranki** [comment]
Sorry for slow reply, I've been on vacation. 

What we need to do is: 
 - Add a boolean to server side thread object to determine if it's "watched". 
 - Add support to it to protocol to change this bit during sync.
 - Figure out a logic on how to handle this bit. What happens if the thread is not found in update? Some time limit perhaps, if the thread is not seen in 4 weeks then forget it?
